rotkehlchen/chain/evm/transactions.py

```diff
@@ -447,6 +460,72 @@ def _get_erc20_transfers_for_ranges(
                 queried_ranges=[(start_ts, end_ts)],
             )
 
+    def address_has_been_spammed(self, address: ChecksumEvmAddress) -> bool:
+        """
+        Queries erc20 tranfers for the given address and if it has only spam assets or ignored
+        assets we return True.
+        If any transfer had an unknown asset
+        """
+        start_ts, end_ts = Timestamp(0), ts_now()
+        checekd_tokens = set()
+        with self.database.conn.read_ctx() as cursor:
+            ignored_assets = self.database.get_ignored_asset_ids(cursor)
+
+        log.debug(f'Address detection: querying {self.evm_inquirer.chain_name} ERC20 Transfers for {address} -> {start_ts} - {end_ts}')  # noqa: E501
+        try:
+            for erc20_tx_hashes in self.evm_inquirer.etherscan.get_token_transaction_hashes(
+                account=address,
+                from_ts=start_ts,
+                to_ts=end_ts,
```

Comment:
A note. This checks every 10 transactions. A hard-coded choice for all of our etherscan queries. It's the variable `TRANSACTIONS_BATCH_NUM`.

But perhaps in this particular case since we want to exit as early as possible smaller value would have been better. But also perhaps this is overthinking it.